force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -124,18 +139,19 @@ public with sharing class ClassesObjectsMethods {
     public static String findTheScientist() {
         // The list of scientists' names
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
+        String allScientistsWithAG = '';
 
-        // The variable to store the concatenated string
-
-        // Loop through the list of scientists' names
-
-            // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
+        // Loop through scientistNames and adds the name to the string if it contains the letter 'G'
+        for (String name : scientistNames) {
+            if (name.containsAny('Gg')){
+               allScientistsWithAG += name + ',';
+            }
+        }
```

Comment:
You can collect a List of scientistNames and then .join it together with a , to make a CSV. No need to remove training ,